SimTracker/TrackAssociatorProducers/plugins/QuickTrackAssociatorByHitsImpl.cc

```diff
@@ -402,33 +400,56 @@ template<typename iter> std::vector<OmniClusterRef> QuickTrackAssociatorByHitsIm
     if (rhit->isValid()) {
       int subdetid = rhit->geographicalId().subdetId();
       if (subdetid==PixelSubdetector::PixelBarrel||subdetid==PixelSubdetector::PixelEndcap) {
-	const SiPixelRecHit* pRHit = dynamic_cast<const SiPixelRecHit*>(rhit);
-	if (!pRHit->cluster().isNonnull())
-	  edm::LogError("TrackAssociator") << ">>> RecHit does not have an associated cluster!" << " file: " << __FILE__ << " line: " << __LINE__;
-	returnValue.push_back(pRHit->omniClusterRef());
+	const std::type_info &tid = typeid(*rhit);
+        LogDebug("TrackAssociator") << "rhit type " << typeid(*rhit).name();
+	if (tid == typeid(SiTrackerMultiRecHit)) {
+          LogDebug("TrackAssociator") << "multi rechit associated!"; 
+	  const SiTrackerMultiRecHit* mHit = dynamic_cast<const SiTrackerMultiRecHit*>(rhit);
+          rhit = getMostProbableHit(mHit);
+        }
```

Comment:
Lines 405-409 are the same as lines 425-429. You can do this part once commonly for pixel+strip (right after checking `rhit` validity), and maybe as
```
if(typeid(*rhit) == typeid(SiTrackerMultiRecHit))
```
to not need both `tid` and `tid2`.

Reply:
This is a great idea, thanks!